CLI use managment and API endpoints

pascal260303 opened the pull request:
I added the package `dj-rest-auth` which provides REST API endpoints for login, logout and more.\
It would also be really easy to later add API endpoints for registration with this.

To manage users with the CLI I added the commands `user add`,`user remove`, `user change-password` and `user list` which are all documented in the [CLI documentation](https://github.com/brendel-group/mission_db/blob/feature/cli-user-management/docs/cli/README.md).

The passwords are validated using some rules set in settings.py in `AUTH_PASSWORD_VALIDATORS` I made a summary of these rules [here](https://github.com/brendel-group/mission_db/blob/feature/cli-user-management/docs/password_policy/README.md) (these are the default rules by django).

The password input is hidden and not stored in any history file.

The passwords are stored in the database as hash values. You can directly see how they are stored with the `user list` command of the CLI.

I made a short summary of the available REST API endpoints in the [restapi documentation](https://github.com/brendel-group/mission_db/blob/feature/cli-user-management/docs/restapi/README.md) and added a link to the complete documentation of `dj-rest-auth`.

I added tests for the CLI but not for the REST API since the endpoints are provided by a package and there is no implementation from our side, so I don't think we need tests for that.

It's possible to have users with an email but it's not required and not used currently. It would be possible to add password reset for forgotten passwords with the email, the API endpoints for this are already there but for this the server has to be able to send emails.

resolves #134 

## How to test
Create a user with the CLI and try to change the password, list and remove the user.\
Test the API endpoints with the created user and check the documentation for errors.

DerSimi reviewed (changes requested):
Hi Pascal,
went through your changes, adding, removing, listing and changing the password of a user works as expected. However, you missed cases like:

"user remove": User 'None' does not exist.
Or 
"user add": Does asks for entering a password, at the end of the dialogue, an exception is thrown. WIth change-password without name argument, the password of user None can be changed.

Regarding the frontend site, i will look into it in another PR. :)

pascal260303 commented:
You're right I forget these cases.\
I now added username validation before asking for a password and made the argument `--name` required for all added commands with that argument.\
Changing the password of the user None was actually not possible it just asked for a password, before it was checked if a user with that name exists.\
`add user` without a name or `None` is not possible, and the exception was thrown because it tried to delete the user because of another exception, but the user didn't exist. This is also now fixed.

> "user remove": User 'None' does not exist.

This is correct, because the user with no name was never created.
